src/module_PI/Raspberry_PI/JUnitTests/DataPacketTest.java

```diff
@@ -2,33 +2,117 @@
 
 import static org.junit.Assert.*;
 
+import java.util.Arrays;
+
 import org.junit.Test;
 
+import module_PI.Raspberry_PI.main.DataPacket;
+import module_PI.Raspberry_PI.main.PacketLogLevel;
+import module_PI.Raspberry_PI.main.PacketType;
+
 public class DataPacketTest {
 
 	@Test
 	public void testDataPacketInt() {
-		fail("Not yet implemented"); // TODO
+		int packetInt = 54258;
+		DataPacket packet = new DataPacket(packetInt);
+		assertEquals(packet.getPacketAsInt(), packetInt);
+	}
+	
+	@SuppressWarnings("unused")
+	@Test(expected = IllegalArgumentException.class)
+	public void testDataPacketInt_invalidLength_tooLong(){
+		DataPacket zeroLevelPacket = new DataPacket(505425);
+	}	
+	
+	@SuppressWarnings("unused")
+	@Test(expected = IllegalArgumentException.class)
+	public void testDataPacketInt_invalidLength_tooShort(){
+		DataPacket zeroLevelPacket = new DataPacket(5425);
+	}	
+	
+	@SuppressWarnings("unused")
+	@Test(expected = IllegalArgumentException.class)
+	public void testDataPacketInt_zeroLogLevel(){
+		DataPacket zeroLevelPacket = new DataPacket(50542);
+	}
+	
+	@SuppressWarnings("unused")
+	@Test(expected = IllegalArgumentException.class)
+	public void testDataPacketInt_highLogLevel(){
+		DataPacket zeroLevelPacket = new DataPacket(56542);
 	}
 
+	@SuppressWarnings("unused")
+	@Test(expected = IllegalArgumentException.class)
+	public void testDataPacketInt_zeroPacketType(){
+		DataPacket zeroPacketTypePacket = new DataPacket(05364);
+	}
+	
+	@SuppressWarnings("unused")
+	@Test(expected = IllegalArgumentException.class)
+	public void testDataPacketInt_highPacketType(){
+		DataPacket zeroPacketTypePacket = new DataPacket(65364);
+	}
+
+	
+	@Test
+	public void testDataPacketByteArrayFromByteArray() {
+		byte[] packetArray = {1,2,3,4,5};
+		DataPacket packet = new DataPacket(packetArray);
+		assertTrue(Arrays.equals(new byte[] {1,2,3,4,5}, packet.getPacketAsArray()));	
+	}
+	
+	@SuppressWarnings("unused")
+	@Test(expected = IllegalArgumentException.class)
+	public void testDataPacketByteArrayFromByteArray_invalidLength() {
+		byte[] packetArray = {1,2,3,4,5,6};
```

Comment:
You might also want to test sending an empty array. Sometimes empty collections cause different errors to occur than this one that you're catching.